src/main/java/it/gov/pagopa/payhub/activities/service/JaxbTrasformerService.java

```diff
@@ -0,0 +1,61 @@
+package it.gov.pagopa.payhub.activities.service;
+
+import it.gov.pagopa.payhub.activities.exception.ActivitiesException;
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBElement;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.Unmarshaller;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import org.xml.sax.SAXException;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+
+/**
+ * A service for transforming XML files into Java objects using JAXB.
+ * <p>
+ * This service supports optional validation of the XML against an XSD schema.
+ * It is lazily loaded to optimize startup time and uses SLF4J for logging.
+ * </p>
+ */
+@Lazy
+@Slf4j
+@Service
+public class JaxbTrasformerService {
+
+	/**
+	 * Unmarshals an XML file into a Java object of the specified type.
+	 *
+	 * @param <T>   the type of the resulting Java object
+	 * @param file  the XML file to parse
+	 * @param clazz the class type to which the XML should be unmarshalled
+	 * @param xsd   the URL of the XSD schema to validate the XML against; if {@code null}, no validation is performed
+	 * @return the unmarshalled Java object of type {@code T}
+	 * @throws ActivitiesException if an error occurs during parsing or validation
+	 */
+	public <T> T unmarshaller(File file, Class<T> clazz, URL xsd) {
+		try (InputStream is = new FileInputStream(file)) {
+			Unmarshaller unmarshaller = JAXBContext.newInstance(clazz).createUnmarshaller();
```

Comment:
`JAXBContext.newInstance(clazz)`
JAXBContext is a thread safe object, whose build could be an heavy operation.

Also the Schema is thread safe.

Thus, we could modify this method in order to have a JAXBContext and Schema instances specialized on the input class and xsd.

We have 2 choice where to store these instances variables:
1. On a separate UnmarshallerService for each XSD
2. On the service specialized on the handling of the file, which will have the XSD url as input